CMakeLists.txt

```diff
@@ -171,6 +171,11 @@ if (ENABLE_NEW_GAIN_BEHAVIOUR)
     add_definitions(-DNEW_GAIN_BEHAVIOUR)
 endif()
 
+if (CMAKE_SYSTEM_NAME MATCHES "BSD")
+    include_directories("/usr/local/include")
```

Comment:
Hello @cederom , could you provide more details about the issue you faced without this line?

Reply:
Hello and thank you @Gluttton for your time :-) In FreeBSD architecture the /bin and /usr/bin locations are reserved for base system only, nothing can change there except things that are provided as part of release and its updates, all user related stuff goes to /usr/local/ including userland binaries configurations libraries and includes, so we need to provide this additional path in order to build package successfully :-)

Reply:
Thank you for the clarification. I haven't had an experience with FreeBSD and I didn't know about this feature.

I asked for more details to better understand either the proposed solution is a proper fix or a workaround. To be frankly, to me adding a hardcoded include path into `CMakeLists.txt` looks like a workaround.  Based on my personal experience with CMake under Linux, if the required header files are placed in a non-standard path they may be found by calling a proper `found_package` directive inside `CMakeLists.txt` or via passing `CMAKE_PREFIX_PATH` via the command line. So I would suggest not adding a hardcoded path into `CMakeLists.txt`.

Anyway the final word on LimeSDR maintainers.

P.S. By the way, seems like the issue has been already fixed in CMake v3.18.2: https://gitlab.kitware.com/cmake/cmake/-/issues/21117 .

Reply:
No worries if there is a better solution we should use a better solution! :-)

`cmake` on FreeBSD is `3.19.2` at the moment. Can you please suggest a more elegant solution on how to add `/usr/local/include` path to all CMake configurations?

Here is our manual on how to use CMake with FreeBSD Ports (automated way to fetch and build a program along with all its dependencies.. Gentoo Portege is based on FreeBSD Ports):
https://www.freebsd.org/doc/en_US.ISO8859-1/books/porters-handbook/building.html#using-cmake

Reply:
`/usr/local` is not really a non-standard path, it was here before Linux was even born, maybe this is more about Linux compatibility with the rest of the world :-) Not sure why CMake does not search there. Maybe there is another place where all search paths are defined? Maybe patching CMake upstream could be the solution? :-)
I have checked `CMAKE_PREFIX_PATH` and this variable is not supported in FreeBSD Ports (yet). Maybe we could add it to solve path problem for all ports. I wrote do our mailing lists maybe someone smart with cmake already know the solution :-)

Reply:
CMake 3.19 documentation uses `/usr/local` as default value of `CMAKE_INSTALL_PREFIX`:
https://cmake.org/cmake/help/v3.19/variable/CMAKE_INSTALL_PREFIX.html

`CMAKE_PREFIX_PATH` is empty by default, so the project sets the value, so the `/usr/local` is missing now and should be put here (but still in the source code so the patch seems unavoidable):
https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html?highlight=cmake_prefix_path

Reply:
> I have checked CMAKE_PREFIX_PATH and this variable is not supported in FreeBSD Ports (yet).

Yes, as I can see that FreeBSP supports a small number of custom variables for port building with CMake. Not sure, but in my opinion the `CONFIGURE_ENV` looks like suitable for passing any parameters for CMake.

> CMake 3.19 documentation uses /usr/local as default value of CMAKE_INSTALL_PREFIX:
> https://cmake.org/cmake/help/v3.19/variable/CMAKE_INSTALL_PREFIX.html

Please note, since `/usr/local` is a part of `CMAKE_INSTALL_PREFIX` it also should be available in the `CMAKE_SYSTEM_PREFIX_PATH` (https://cmake.org/cmake/help/v3.19/variable/CMAKE_SYSTEM_PREFIX_PATH.html), so the `/usr/local` should be available by default without additional passing via command line parameters.

Could you provide build error output? Probably it helps to fix the issue.

Reply:
Thanks @Gluttton for the reference [1]. It looks like problem is somewhere with `find_package`, `find_library`, `find_file`, `find_path` and this is the reason why some GL headers are not found/set. Will look into this in a free moment. If you find a better solution first please let us know :-) Thanks :-)

https://gitlab.kitware.com/cmake/cmake/-/issues/21117

Reply:
```
FAILED: src/CMakeFiles/LimeSuiteGUI.dir/fftviewer_wxgui/fftviewer_frFFTviewer.cpp.o
/usr/bin/c++ -DLIMERFE -DLIME_DLL -DWIN32_LEAN_AND_MEAN -DWXUSINGDLL -D_FILE_OFFSET_BITS=64 -D_THREAD_SAFE -D__WXGTK__ -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src
/lms7002_wxgui -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/LMS_Programing -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/utilities_gui -I/usr/ports/comms
/limesuite/work/LimeSuite-20.10.0/src/RFSpark -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/FPGAcontrols_wxgui -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/s
rc/numericSlider -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/fftviewer_wxgui -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/lms7suiteEvents -I/usr/ports/
comms/limesuite/work/LimeSuite-20.10.0/src/boards_wxgui -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/limeRFE -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/sr
c -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/API -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/lime -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0
/src/GFIR -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/protocols -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/ConnectionRegistry -I/usr/ports/comms/lime
suite/work/LimeSuite-20.10.0/src/lms7002m_mcu -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/ADF4002 -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/Si5351C
-I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/lms7002m -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/FPGA_common -I/usr/ports/comms/limesuite/work/LimeSui
te-20.10.0/external/cpp-feather-ini-parser -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/external/kissFFT -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/thread
Helper -I/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/oglGraph -isystem /usr/local/lib/wx/include/gtk3-unicode-3.1 -isystem /usr/local/include/wx-3.1 -pipe -g -O0 -f
stack-protector-strong -fno-strict-aliasing -pthread -pipe -g -O0 -fstack-protector-strong -fno-strict-aliasing -pthread -std=gnu++11 -MD -MT src/CMakeFiles/LimeSuiteGUI.dir/
fftviewer_wxgui/fftviewer_frFFTviewer.cpp.o -MF src/CMakeFiles/LimeSuiteGUI.dir/fftviewer_wxgui/fftviewer_frFFTviewer.cpp.o.d -o src/CMakeFiles/LimeSuiteGUI.dir/fftviewer_wxg
ui/fftviewer_frFFTviewer.cpp.o -c /usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/fftviewer_wxgui/fftviewer_frFFTviewer.cpp
In file included from /usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/fftviewer_wxgui/fftviewer_frFFTviewer.cpp:4:
In file included from /usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/oglGraph/OpenGLGraph.h:10:
/usr/ports/comms/limesuite/work/LimeSuite-20.10.0/src/oglGraph/glew/GL/glew.h:1202:14: fatal error: 'GL/glu.h' file not found
#    include <GL/glu.h>
             ^~~~~~~~~~
1 error generated.
```

Reply:
I'v found the related topic on SO: https://stackoverflow.com/questions/32624212/cmake-on-freebsd-doesnt-see-gl-gl-h-in-usr-local-include . According to the topic to fix the issue it is necessary to call `find_package` for OpenGL and then explicitly add `${OPENGL_INCLUDE_DIR}` into include directories.

I've checked and LimeSuite doesn't contain `find_package` for OpenGL.
```
LimeSuite$ find . -name 'CMakeLists.txt' -exec grep -n 'find_package' {} +
./CMakeLists.txt:226:find_package(Doxygen QUIET)
./QuickTest/CMakeLists.txt:25:    find_package(FLTK)
./src/CMakeLists.txt:111:find_package(Threads REQUIRED)
./src/CMakeLists.txt:176:find_package(wxWidgets COMPONENTS richtext xml html xrc adv aui core base gl)
./src/ConnectionSTREAM_UNITE/CMakeLists.txt:6:#    find_package(CyAPI)
./src/ConnectionSTREAM_UNITE/CMakeLists.txt:11:#    find_package(libusb-1.0)
./src/examples/CMakeLists.txt:10:find_package(Gnuplot)
./src/ConnectionFX3/CMakeLists.txt:6:    find_package(CyAPI)
./src/ConnectionFX3/CMakeLists.txt:11:    find_package(libusb-1.0)
./src/ConnectionFTDI/CMakeLists.txt:33:    find_package(libusb-1.0)
./SoapyLMS7/CMakeLists.txt:5:find_package(SoapySDR CONFIG)
```

Could you add changes mentioned in the SO topic and check building on your environment again?